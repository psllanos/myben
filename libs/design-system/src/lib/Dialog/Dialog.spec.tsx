import { render, screen } from '@testing-library/react'
import { Dialog } from './'

describe('Dialog', () => {
    describe('when rendered with `open` prop true', () => {
        it('should display the modals contents', () => {
            const onToggleMock = jest.fn()
            const component = render(
                <Dialog isOpen={true} onClose={onToggleMock}>
                    <Dialog.Title>Dialog Title</Dialog.Title>
                    <Dialog.Content>Dialog Content</Dialog.Content>
                    <Dialog.Description>Dialog Description</Dialog.Description>
                    <Dialog.Actions>Dialog Actions</Dialog.Actions>
                </Dialog>
            )

            expect(screen.getByText('Dialog Title')).toBeInTheDocument()
            expect(screen.getByText('Dialog Content')).toBeInTheDocument()
            expect(screen.getByText('Dialog Description')).toBeInTheDocument()
            expect(screen.getByText('Dialog Actions')).toBeInTheDocument()
            expect(component).toMatchSnapshot()
        })
    })
})
