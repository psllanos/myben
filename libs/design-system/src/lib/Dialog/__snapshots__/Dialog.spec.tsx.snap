// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog when rendered with \`open\` prop true should display the modals contents 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
      />
    </div>
    <div
      id="headlessui-portal-root"
    >
      <div
        data-headlessui-portal=""
      >
        <button
          aria-hidden="true"
          style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
          type="button"
        />
        <div>
          <div
            aria-describedby="headlessui-description-:r3:"
            aria-labelledby="headlessui-dialog-title-:r2:"
            aria-modal="true"
            class="relative z-10"
            data-headlessui-state="open"
            id="headlessui-dialog-:r0:"
            role="dialog"
          >
            <div
              aria-hidden="true"
              class="fixed inset-0 bg-gray-800 bg-opacity-80 transition-opacity"
            />
            <div
              class="fixed z-20 inset-0 overflow-y-auto"
            >
              <div
                class="flex items-center justify-center p-4 sm:p-0 min-h-full text-center "
              >
                <div
                  class="relative p-4 sm:p-6 sm:my-8 w-full bg-gray-700 rounded text-left shadow-md shadow-black transform transition-all sm:max-w-md"
                  data-headlessui-state="open"
                  id="headlessui-dialog-panel-:r1:"
                >
                  <div
                    class="w-full flex items-start justify-between"
                  >
                    <h4
                      data-headlessui-state="open"
                      id="headlessui-dialog-title-:r2:"
                    >
                      Dialog Title
                    </h4>
                    <div
                      class="shrink-0 pl-6 ml-auto"
                    >
                      <button
                        class="h-8 w-8 flex items-center justify-center bg-transparent text-gray-50 hover:bg-gray-500 rounded focus:bg-gray-400 focus:outline-none"
                        type="button"
                      >
                        <svg
                          class="h-6 w-6"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g>
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636z"
                            />
                          </g>
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="mt-6"
                  >
                    <div>
                      Dialog Content
                    </div>
                    <p
                      class="text-base text-gray-50 undefined"
                      data-headlessui-state="open"
                      id="headlessui-description-:r3:"
                    >
                      Dialog Description
                    </p>
                  </div>
                  <div
                    class="flex space-x-2 mt-6 undefined"
                  >
                    Dialog Actions
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          aria-hidden="true"
          style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
          type="button"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
